opentelemetry-sdk/tests/performance/benchmarks/trace/test_benchmark_trace.py

```diff
@@ -0,0 +1,51 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import opentelemetry.sdk.trace as trace
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import sampling
+
+tracer = trace.TracerProvider(
+    sampler=sampling.DEFAULT_ON,
+    resource=Resource(
+        {
+            "service.name": "A123456789",
+            "service.version": "1.34567890",
+            "service.instance.id": "123ab456-a123-12ab-12ab-12340a1abc12",
+        }
+    ),
+).get_tracer("sdk_tracer_provider")
+
+
+def test_simple_start_span(benchmark):
```

Comment:
What in here constitutes this as  benchmark test? How does pytest-benchmark know?

Reply:
I had to do some reading on this myself, it's literally the parameter named `benchmark`. It know to look for this **pytest fixture** because we downloaded `pytest-benchmark`. If you call it `foo` you'd get an error like this:

```
collected 1 item

performance/benchmarks/test_benchmark_prime_factors.py E                                                                                              [100%]

========================================================================== ERRORS ===========================================================================
__________________________________________________ ERROR at setup of test_instrumentation_1_prime_factors ___________________________________________________
file /Users/enowell/git/github-benchmarks-example/instrumentation-1/tests/performance/benchmarks/test_benchmark_prime_factors.py, line 3
  def test_instrumentation_1_prime_factors(foo):
E       fixture 'foo' not found
>       available fixtures: benchmark, benchmark_weave, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, monkeypatch, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/Users/enowell/git/github-benchmarks-example/instrumentation-1/tests/performance/benchmarks/test_benchmark_prime_factors.py:3
================================================================== short test summary info ==================================================================
ERROR performance/benchmarks/test_benchmark_prime_factors.py::test_instrumentation_1_prime_factors
===================================================================== 1 error in 0.07s ======================================================================
ERROR: InvocationError for command /Users/enowell/git/github-benchmarks-example/.tox/test-instrumentation-1/bin/pytest (exited with code 1)
__________________________________________________________________________ summary __________________________________________________________________________
ERROR:   test-instrumentation-1: commands failed
```

You can see the output tells you which fixtures are available.

More reading [here on the pytest-benchmark documentation](https://pytest-benchmark.readthedocs.io/en/latest/usage.html#usage)

Reply:
Awesome. This was what I was referring to when I said we should have something in `CONTRIBUTING.md` to help users write benchmark tests properly. A link to the docs and a description saying "we need to have benchmark parameter" is sufficient I feel.